src/components/ui/decimalsInput.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from "react";
+import { Input, InputProps } from "./input";
+
+interface DecimalsInputProps extends Omit<InputProps, "onChange" | "value"> {
+  value: string;
+  onChange: (value: string) => void;
+  decimals?: number; // Decimal limit, defaults to 18
+}
+
+const DecimalsInput: React.FC<DecimalsInputProps> = ({ value, onChange, decimals = 18, ...props }) => {
+  const decimalsRegex = new RegExp(`^\\d*\\.?\\d{0,${decimals}}$`);
```

Comment:
Good thinking! Although right now, the user won't get feedback right?

Couple ways we can go about it:

1. Wrap the PayForm content in an actual `form`, and do form validation with [zod](https://zod.dev/) and a form lib.
2. Maybe simpler for now: add a `pattern` prop to the input with the decimal regex, and add a `onInvalid` callback to set the msg

Let me know if this doesnt makes sense

Reply:
I think the current solution is pretty simple and good – it's like the default behavior for the browser's `type=number` input, except it tracks the value as a string internally meaning it can be used up to 18 decimals. Maybe not great for a11y.

If you'd like, I can add a little error msg beneath it when the user fails the regexp which gets cleared on new inputs.

Option 2 sounds good as well! 1 maybe a bit overkill imo.

Reply:
added a simple warning message!